code/datums/components/machine_power_modifier.dm

```diff
@@ -0,0 +1,45 @@
+/datum/component/machine_power_modifier
+	dupe_mode = COMPONENT_DUPE_ALLOWED
+	/// Multiplier for the idle power consumption of the attached machine.
+	var/idle_power_multiplier = 1
+	/// Multiplier for the active power consumption of the attached machine.
+	var/active_power_multiplier = 1
+	/// Trait to give to the machine we're getting attached to.
+	var/trait_to_add
+
+	var/list/objects_to_drop
+
+/datum/component/machine_power_modifier/Initialize(idle_power_multiplier = 1, active_power_multiplier = 1, trait_to_add, has_deconstruction_property, list/objects_to_drop)
```

Comment:
Instead of needing `has_deconstruction_property`, just check that any `objects_to_drop` were passed